src/Auth0/Login/Auth0Service.php

```diff
@@ -167,7 +173,28 @@ public function rememberUser($value = null)
      */
     public function decodeJWT($encUser, array $verifierOptions = [])
     {
-        $this->apiuser = $this->auth0->decodeIdToken($encUser, $verifierOptions);
+        $token_issuer = 'https://'.$this->auth0Config['domain'].'/';
+        $apiIdentifier = $this->auth0Config['api_identifier'];
+        $idTokenAlg = $this->auth0Config['supported_algs'][0] ?? 'RS256';
+
+        $signature_verifier = null;
+        if ('RS256' === $idTokenAlg) {
+            $jwksUri = $this->auth0Config['jwks_uri'] ?? 'https://'.$this->auth0Config['domain'].'/.well-known/jwks.json';
+            $jwks_fetcher = new JWKFetcher($this->auth0Config['cache_handler']);
+            $jwks = $jwks_fetcher->getKeys($jwksUri);
+            $signature_verifier = new AsymmetricVerifier($jwks);
+        } else if ('HS256' === $idTokenAlg) {
+            $signature_verifier = new SymmetricVerifier($this->auth0Config['client_secret']);
+        }
+
+        // Use IdTokenVerifier since Auth0-issued JWTs contain the 'sub' claim, which is used by the Laravel user model
+        $token_verifier = new IdTokenVerifier(
```

Comment:
Thanks for the background @joshcanhelp, that makes sense.

I did pull down https://github.com/auth0/auth0-PHP/pull/434, and while it will get past the missing nonce failure if we pass the options `["nonce" => false]` when calling `decodeIdToken`, it will fail on the audience validation as `decodeIdToken` constructs the token verifier using the `clientId` instead of the audience.

But given that we should use `TokenVerifier` instead of `IdTokenVerifier` for the reasons you listed above, it seems we either still need to do the validation here in `laravel-auth0` (but change to use `TokenVerifier`, or provide another API or configuration point to allow decoding/validating of Auth0-issued JWT access tokens. What do you think the best approach is?

Reply:
Updated this PR to use `TokenVerifier` instead of `IdTokenVerifier` per the discussion above and after discussing with @joshcanhelp